### PR TITLE
Reusable result frame

### PR DESCRIPTION
**Pull request by OskarDamkjaer**

### CODE REVIEW

File: src/browser/modules/Editor/EditorFrame.tsx [hunk 1]
```diff
@@ -39,7 +39,6 @@ import {
   REMOVE_FAVORITE,
   updateFavorite
 } from 'shared/modules/favorites/favoritesDuck'
-import { useSpring, animated } from 'react-spring'
```
Comment (OskarDamkjaer) [inline src/browser/modules/Editor/EditorFrame.tsx:42]: Removed unneeded animation and dependency on react-spring

File: build_scripts/webpack-plugins.js [hunk 2]
```diff
@@ -84,15 +84,15 @@ module.exports = () => {
...
-    new ForkTsCheckerWebpackPlugin({
-      eslint: {
-        files: './src/**/*.{ts,tsx,js,jsx}'
-      }
-    }),
-    new ForkTsCheckerNotifierWebpackPlugin({
-      title: 'TypeScript',
-      excludeWarnings: false
-    }),
+    //new ForkTsCheckerWebpackPlugin({
```
Comment (HerrEmil) [inline build_scripts/webpack-plugins.js]: revert

File: src/browser/modules/Frame/FrameTitlebar.tsx [hunk 3]
```diff
@@ -77,13 +77,69 @@ import {
...
+  runQuery: () => any
+  bus: Bus
+}
+
+type FrameTitleBarProps = FrameTitleBarBaseProps & {
+  request: Request | null
+  isRelateAvailable: boolean
+  newFavorite: (cmd: string) => void
+  newProjectFile: (cmd: string) => void
+  onCloseClick: (a: any, b: any, c: any) => void
```
Comment (HerrEmil) [inline src/browser/modules/Frame/FrameTitlebar.tsx]: naming

File: src/browser/modules/Frame/FrameTitlebar.tsx [hunk 4]
```diff
@@ -77,13 +77,69 @@ import {
...
+  frame contents.
+  */
 
-class FrameTitlebar extends Component<any> {
-  hasData() {
-    return this.props.numRecords > 0
+  const gainFocusCallback = useCallback(() => {
+    if (props.frame.isRerun) {
+      editorRef.current?.focus()
+      editorRef.current?.setPosition({ lineNumber: 999, column: 999 })
```
Comment (HerrEmil) [inline src/browser/modules/Frame/FrameTitlebar.tsx]: set last

File: src/browser/modules/Frame/FrameTitlebar.tsx [hunk 5]
```diff
@@ -129,178 +185,189 @@ class FrameTitlebar extends Component<any> {
...
           >
-            {expandCollapseIcon}
+            {fullscreenIcon}
           </FrameButton>
-          <Render if={frame.type === 'edit'}>
-            <FrameButton title="Run" onClick={() => props.onRunClick()}>
-              <SVGInline svg={controlsPlay} width="12px" />
-            </FrameButton>
-          </Render>
-          <Render if={frame.type !== 'edit'}>
```
Comment (HerrEmil) [inline src/browser/modules/Frame/FrameTitlebar.tsx:280]: Unbreak or remove

File: src/browser/modules/Frame/styled.tsx [hunk 6]
```diff
@@ -187,19 +159,41 @@ export const StyledFrameTitlebarButtonSection = styled.ul`
...
+  color: ${props => props.theme.promptText};
+`
+
+export const FrameTitleEditorContainer = styled.div`
+  border-radius: 2px;
+  padding-left: 6px;
+  padding-top: 3px;
+  margin: 3px 5px 3px 3px;
+
+  width: 0; // Prevents the editor from growing past flex-grow: 1
```
Comment (HerrEmil) [inline src/browser/modules/Frame/styled.tsx:182]: safari?

File: src/browser/modules/Stream/stream.utils.ts [hunk 7]
```diff
@@ -18,7 +18,9 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-export function getLatestFromFrameStack(frameObj: any) {
+import { Frame, FrameStack } from 'shared/modules/stream/streamDuck'
+
+export function getLatestFromFrameStack(frameObj: FrameStack): Frame | null {
```
Comment (HerrEmil) [inline src/browser/modules/Stream/stream.utils.ts:23]: unused?

File: src/shared/modules/commands/commandsDuck.test.ts [hunk 8]
```diff
@@ -156,6 +156,7 @@ describe('commandsDuck', () => {
             params: { x: 2 }
           } as any),
           updateQueryResult(
+            // @ts-expect-error
```
Comment (HerrEmil) [inline src/shared/modules/commands/commandsDuck.test.ts]: ???

File: src/shared/modules/stream/streamDuck.ts [hunk 9]
```diff
@@ -61,15 +61,20 @@ export function getRecentView(state: GlobalState): null | FrameView {
...
   if (newState.parentId && state.allIds.indexOf(newState.parentId) < 0) {
-    // No parent
+    // Can't find parent
     return state
   }
 
   const frameObject = state.byId[newState.id] || { stack: [], isPinned: false }
-  if (!newState.isRerun) {
-    frameObject.stack.unshift(newState)
+  // shoehorned history for re-usable frame. Should be kept in local state after refractor
```
Comment (HerrEmil) [inline src/shared/modules/stream/streamDuck.ts]: redux > local

File: src/browser/modules/Frame/styled.tsx (hunk 6 above)
Comment (OskarDamkjaer) [reply src/browser/modules/Frame/styled.tsx:182]: Tested and it works

File: src/browser/modules/Stream/stream.utils.ts (hunk 7 above)
Comment (OskarDamkjaer) [reply src/browser/modules/Stream/stream.utils.ts:23]: Used in commandInterpreterHelper

**Pull request closed without merging**